### DISCUSSION TITLE
What alternatives exist to deprecated any() and all() functions?

### DISCUSSION THREAD

**Discussion by anderseknert** [open-policy-agent/community#108] [OPA and Rego]
OPA used to include functions called `any` and `all`, but these have been deprecated for a long time. Since people sometimes see these in older versions of the docs and wonder what they were doing, or why they were deprecated, here's an explanation.

**any**
`output := any(array)` output would be `true` if any of the elements in `array` was equal to `true`

This is trival to do by other means. Simply use the `in` operator to check for presence of `true` in an array.

```rego
import future.keywords

allow {
    true in array
}
```

Using `in` of course has the benefit that it can check for **any** type of value, not just `true`. 

**all**
`output := all(array)` output would be `true` if all of the elements in `array` was equal to `true`. This can easily be checked using `every`

```rego
import future.keywords

allow {
    every x in array { x == true }
}
```

If you want to replace the deprecated functions with your own implementations, you could use something like the below:

```rego
package play

import future.keywords

any(arr) := true in arr
all(arr) {
    every x in arr {x == true}
}
```

**Comment by peteroneilljr** [comment]
@anderseknert might be time to update this to show the `every` keyword.

**Reply by anderseknert** [reply]
Thanks! Done.
